### PR TITLE
tcp/udp_connect: If the remote addr is ANY, change it to LOOPBACK

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Behavior alignment to Linux

## Impact

connect

### CODE REVIEW

File: net/tcp/tcp_conn.c [hunk 1]
```diff
@@ -1242,7 +1242,14 @@ int tcp_connect(FAR struct tcp_conn_s *conn, FAR const struct sockaddr *addr)
...
 
-      net_ipv4addr_copy(conn->u.ipv4.raddr, inaddr->sin_addr.s_addr);
+      if (inaddr->sin_addr.s_addr == INADDR_ANY)
+        {
+          net_ipv4addr_copy(conn->u.ipv4.raddr, HTONL(INADDR_LOOPBACK));
+        }
+      else
+        {
+          net_ipv4addr_copy(conn->u.ipv4.raddr, inaddr->sin_addr.s_addr);
+        }
```
Comment (acassis) [inline net/tcp/tcp_conn.c:1252]: So, 0.0.0.0 means 127.0.0.1? Is that the idea? Maybe should be a good idea include a comment.

Comment (xiaoxiang781216) [reply net/tcp/tcp_conn.c:1252]: Here has some dissussion:
https://unix.stackexchange.com/questions/419880/connecting-to-ip-0-0-0-0-succeeds-how-why

Comment (xiaoxiang781216) [reply net/tcp/tcp_conn.c:1254]: Done.

**Review by anchao — APPROVED** [review]
LGTM

**Pull request merged**